core/darwin/src/Converters.kt

```diff
@@ -82,3 +91,12 @@ public fun LocalDateTime.toNSDateComponents(): NSDateComponents {
     components.nanosecond = nanosecond.convert()
     return components
 }
+
+/**
+ * Converts the given [LocalDateTime] to [NSDate].
+ */
+public fun LocalDateTime.toNSDate(): NSDate? {
```

Comment:
The NSDateComponents convertion to NSDate is not guaranteed.
`components.date` this line returns an optional NSDate.
But I think that if we give the NSDateComponents a NSCalendar, is should always return a valide NSDate, or almost always, because working with dates is not always strait forward.

Reply:
I would think that `components.date` can return `nil` if the calendar is not set or the necessary components are not provided. In any case, we would need to document the specific conditions under which `null` can be returned here.

Reply:
I think you are correct.